src/components/QSnackBar/QSnackBar.vue

```diff
@@ -0,0 +1,235 @@
+    </div>
+  </div>
+</template>
+
+<script>
+
+export default {
+  name: 'QSnackBar',
+  props: {
+    variant: {
+      type: String,
+      default: "tile",
+      validator(value) {
+        const validVariant = ['tile', 'rounded', 'shaped', 'outline'];
+        return validVariant.includes(value);
+      },
+    },
+    position: {
+      type: String,
+      validator(value) {
+        const validPosition = ['top-center', 'bottom-center', 'top-right', 'top-left', 'bottom-left', 'bottom-right'];
+        return validPosition.includes(value);
+      },
+    },
+    open: {
+      type: Boolean,
+      default: false,
+    },
+    colorBackground: {
+      type: String,
+      default: 'primary',
+      validator(value) {
+        const validColorSnackbar = ['primary', 'error', 'success', 'secondary', 'warning'];
+        return validColorSnackbar.includes(value);
+      },
+    },
+    colorButton: {
+      type: String,
+      default: '#ffffff',
+    },
+    colorText: {
+      type: String,
+      default: '#ffffff',
+      validator(value) {
+        const validColorText = ['white', 'black', 'darkgray'];
+        return validColorText.includes(value);
+      },
+    },
+    message: {
+      type: String,
+      default: 'Votre message ici.'
+    },
+  },
+  data() {
+    return {
+      isOpen: this.open
+    }
+  },
+  computed: {
+    allClasses() {
+      const allClasses = [];
+      allClasses.push('snackbar');
+
+      if (this.variant === 'tile') {
+        allClasses.push('snackbar--tile');
+      }
+      if (this.variant === 'rounded') {
+        allClasses.push('snackbar--rounded');
+      }
+      if (this.variant === 'shaped') {
+        allClasses.push('snackbar--shaped');
+      }
+      if (this.variant === 'outline') {
+        allClasses.push('snackbar--outline');
+      }
+
+      if (this.colorBackground === 'primary') {
+        allClasses.push('snackbar--primary');
+      }
+      if (this.colorBackground === 'success') {
+        allClasses.push('snackbar--success');
+      }
+      if (this.colorBackground === 'error') {
+        allClasses.push('snackbar--error');
+      }
+      if (this.colorBackground === 'warning') {
+        allClasses.push('snackbar--warning');
+      }
+      if (this.colorBackground === 'secondary') {
+        allClasses.push('snackbar--secondary');
+      }
+
+      if (this.position === 'top-center') {
+        allClasses.push('snackbar--top-center');
+      }
+      if (this.position === 'top-right') {
+        allClasses.push('snackbar--top-right');
+      }
+      if (this.position === 'top-left') {
+        allClasses.push('snackbar--top-left');
+      }
+      if (this.position === 'bottom-center') {
+        allClasses.push('snackbar--bottom-center');
+      }
+      if (this.position === 'bottom-right') {
+        allClasses.push('snackbar--bottom-right');
+      }
+      if (this.position === 'bottom-left') {
+        allClasses.push('snackbar--bottom-left');
+      }
```

Comment:
why specific to top-center and bottom-center ?

Should be
top
bottom
left
right
and the variantes of them (botton-left, bottom-right, top-left, top-right)